File: spine-libgdx/src/com/esotericsoftware/spine/Event.java

package com.esotericsoftware.spine;

public class Event {
	final private EventData data;
	int intValue;
	float floatValue;
	String stringValue;

	public Event (EventData data) {
		this.data = data;
	}

	public int getInt () {
		return intValue;
	}

	public void setInt (int intValue) {
		this.intValue = intValue;
	}

	public float getFloat () {
		return floatValue;
	}

	public void setFloat (float floatValue) {
		this.floatValue = floatValue;
	}

	public String getString () {
		return stringValue;
	}

	public void setString (String stringValue) {
		this.stringValue = stringValue;
	}

	public EventData getData () {
		return data;
	}

	public String toString () {
		return data.name;
	}
}

File: spine-ts/core/src/Event.ts
module spine {
	export class Event {
		data: EventData;
		intValue: number;
		floatValue: number;
		stringValue: string;
		time: number;

		constructor (time: number, data: EventData) {
			if (data == null) throw new Error("data cannot be null.");
		    this.time = time;
		    this.data = data;
		}
	}
}
